docs/design-survey-conditional-logic.md

```diff
@@ -0,0 +1,691 @@
+
+#### Implement a Custom Function
+
+Your custom function must accept only one array-like parameter that will contain all the passed arguments. For example, the following code passes two arguments to a custom function `myFunc`:
+
+```js
+"expression": "myFunc({question1}, {question2})"
+```
+
+However, the `myFunc` implementation must accept only one parameter&mdash;an array that contains all arguments:
+
+```js
+function myFunc(params) {
+  let q1_value = params[0];
+  let q2_value = params[1];
+  // ...
+  return someValue;
+}
+```
+
+After you implement a custom function, register it in the `FunctionFactory`:
+
+```js
+FunctionFactory.Instance.register("myFunc", myFunc);
+```
+
+For illustrative purposes, the code below shows the built-in `age` function implementation:
+
+```js
+// Accepts a birthdate and returns the current age in full years
+function age(params: any[]): any {
+  if (!params && params.length < 1) return null;
+  if (!params[0]) return null;
+  var birthDate = new Date(params[0]);
+  var today = new Date();
+  var age = today.getFullYear() - birthDate.getFullYear();
+  var m = today.getMonth() - birthDate.getMonth();
+  if (m < 0 || (m === 0 && today.getDate() < birthDate.getDate())) {
+    age -= age > 0 ? 1 : 0;
+  }
+  return age;
+}
+// Register the `age` function under the `age` alias
+FunctionFactory.Instance.register("age", age);
+```
+
+[View source code on GitHub](https://github.com/surveyjs/survey-library/blob/68eb0054dc83d2f45a6daa1042bf7440c8faf007/src/functionsfactory.ts#L218-L230)
+
+#### Access Survey Elements Within a Custom Function
+
+You can access any survey element via a [survey instance](https://surveyjs.io/Documentation/Library?id=surveymodel). Use the `this.survey` property to get the survey instance within your custom function implementation. This property allows you to design your function so that it accepts a survey element name as a parameter:
+
+```js
+"expression": "myFunc('questionName')"
+```
+
+Within the function implementation, you can use the passed name to get the instance of the corresponding element:
+
+```js
+function myFunc(params) {
+  const questionInstance = this.survey.getQuestionByName(params[0]);
+  // ...
+}
+```
+
+#### Asynchronous Functions
+
+If an expression requires time-consuming calculations or a request to a server, implement an asynchronous custom function. The following code shows an example of an asynchronous function:
+
+```js
+function asyncFunc(params: any[]): any {
+  // Preserve the context variable to access the `returnResult` callback later
+  var self = this;
+  setTimeout(function() {
+    // Return the function result via the callback
+    self.returnResult(yourValue);
+  }, 100);
+}
+```
+
+After you implement an asynchronous function, register it in the `FunctionFactory`. The third parameter specifies if this function is asynchronous: 
```

Comment:
After you implement an asynchronous function, register it in `FunctionFactory`. The third parameter specifies if this function is asynchronous: